config/confighttp/confighttp.go

```diff
@@ -237,6 +241,15 @@ func WithErrorHandler(e middleware.ErrorHandler) ToServerOption {
 	}
 }
 
+// WithMaxRecvSize introduce a request body size upper bound for the HTTP handler.
+// This uses http.MaxBytesReader under the hood and when the client sends a request body
+// that exceeds this limit, 400 will be returned.
+func WithMaxRecvSize(maxRecvSize int64) ToServerOption {
```

Comment:
Why public?

Reply:
`WithErrorHandler` is a func pointer, cannot be set in Configuration via yaml, this is not the case here. We don't have similar things for any other part of the config that can be configured in the yaml file.

Reply:
Indeed. I removed the `maxRequestBodySize` field in toServerOptions and the *With* function. Just a `MaxRequestBodySize` field in `HTTPServerSettings`  is already enough.